drivers/wifi/nxp/Kconfig.nxp

```diff
@@ -0,0 +1,1212 @@
+# Copyright 2022-2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig WIFI_NXP
+	bool "NXP Wi-Fi driver support"
+	select THREAD_NAME
+	select THREAD_CUSTOM_DATA
+	select THREAD_LOCAL_STORAGE
+	select NET_SOCKETS
+	select GPIO
+	select NET_L2_WIFI_MGMT
+	select NET_SOCKETPAIR
+	select WIFI_USE_NATIVE_NETWORKING
+	select NET_IPV4
+	select NET_IPV6
+	select NET_DHCPV4
+	select NET_MGMT_EVENT
+	select NET_L2_ETHERNET
+	select NET_L2_ETHERNET_MGMT
+	select POSIX_CLOCK
+	select EVENTS
+	select REQUIRES_FULL_LIBC
+	select SDHC if !SOC_SERIES_RW6XX
+	select SDIO_STACK if !SOC_SERIES_RW6XX
+	imply NET_IPV6_DAD
+	imply NET_IPV6_FRAGMENT
+	imply NET_IPV4_FRAGMENT
+	imply MBEDTLS
+	depends on DT_HAS_NXP_WIFI_ENABLED
+	help
+	  Enable NXP SoC Wi-Fi support.
+
+if WIFI_NXP
+
+module = WIFI_NXP
+
+config WIFI_MGMT_SCAN_CHAN_MAX_MANUAL
+	default 50
+
+config NET_TCP_WORKQ_STACK_SIZE
+	default 2048
+
+config NET_RX_STACK_SIZE
+	default 2048
+
+config NET_MGMT_EVENT_STACK_SIZE
+	default 2048
+
+config SYSTEM_WORKQUEUE_STACK_SIZE
+	default 2048
+
+config SHELL_STACK_SIZE
+	default 5120
+
+config POSIX_MAX_FDS
+	default 12
+
+config SHELL_ARGC_MAX
+	default 32
+
+config NET_BUF_DATA_SIZE
+	default 1744
```

Comment:
If you are planning to use such a large net_buf's, then the PR https://github.com/zephyrproject-rtos/zephyr/pull/71400 could help to lower memory consumption

Reply:
Hi jukkar, thanks for comments. updated all above as commented.
For this one we will take it as optimization to modify later, not in this PR.